### PR TITLE
[Fix] Fix mmseg.api.inference inference_segmentor

### PR DESCRIPTION
**Pull request by jinwonkim93**
## Motivation
This PR aims to solve https://github.com/open-mmlab/mmsegmentation/issues/1643. 
Fix inference_segmentor not working  with multiple images path or images. List[str/ndarray]

## Modification

- process images if instance is list
- change variable img to imgs

### CODE REVIEW

**Comment by jinwonkim93** [comment]
> [CLA assistant check](https://cla-assistant.io/open-mmlab/mmsegmentation?pullRequest=1849) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/open-mmlab/mmsegmentation?pullRequest=1849) before we can accept your contribution.
> 
> **jinwonkim93** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open-mmlab/mmsegmentation?pullRequest=1849) it.

**Comment by MeowZheng** [comment]
Many thanks for your contribution, please sign [CLA](https://cla-assistant.io/open-mmlab/mmsegmentation?pullRequest=1849) first.

**Comment by jinwonkim93** [comment]
> Many thanks for your contribution, please sign [CLA](https://cla-assistant.io/open-mmlab/mmsegmentation?pullRequest=1849) first.

yes, i did it.

**Comment by jinwonkim93** [comment]
@MeowZheng @MengzhangLI Anything you like to comment?

File: mmseg/apis/inference.py [hunk 1]
```diff
@@ -84,9 +84,17 @@ def inference_segmentor(model, img):
...
+        data = []
+        for img in imgs:
+            img_data = dict(img=img)
+            img_data = test_pipeline(img_data)
+            data.append(img_data)
+        data = collate(data, samples_per_gpu=len(imgs))
+    else:
+        data = dict(img=imgs)
+        data = test_pipeline(data)
+        data = collate([data], samples_per_gpu=1)
```
Comment (nijkah) [inline mmseg/apis/inference.py:97]:
```suggestion
    data = []
    imgs = imgs if isinstance(imgs, list) else [imgs]
    for img in imgs:
        img_data = dict(img=img)
        img_data = test_pipeline(img_data)
        data.append(img_data)
    data = collate(data, samples_per_gpu=len(imgs))
```

**Comment by jinwonkim93** [comment]
@MengzhangLI Is there something i could do more?

**Review by nijkah — APPROVED** [review]
LGTM

**Review by MeowZheng — APPROVED** [review]

**Comment by jinwonkim93** [comment]
> 

Thank you for invitation but discord is invalid.

**Comment by jinwonkim93** [comment]
> jinwonkim93，您好！您在MMSeg项目中给我们提的PR非常重要，感谢您付出私人时间帮助改进开源项目，相信很多开发者会从你的PR中受益。 我们非常期待与您继续合作，OpenMMLab专门成立了贡献者组织MMSIG，为贡献者们提供开源证书、荣誉体系和专享好礼，可通过添加微信：openmmlabwx 联系我们（请备注mmsig+GitHub id），由衷希望您能加入！ Dear jinwonkim93, First of all, we want to express our gratitude for your significant PR in the MMSeg project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR. We are looking forward to continuing our collaboration with you. OpenMMLab has established a special contributors' organization called MMSIG, which provides contributors with open-source certificates, a recognition system, and exclusive rewards. You can contact us by adding our WeChat（if you have WeChat): openmmlab_wx, or join in our discord： https://discord.gg/qH9fysxPDW. We sincerely hope you will join us! Best regards！ @jinwonkim93

Thank you for invitation but discord is invalid.

**Pull request merged**